src/NzbDrone.Core/Parser/ParsingService.cs

```diff
@@ -354,91 +342,113 @@ private ParsedEpisodeInfo ParseSpecialEpisodeTitle(string title, Series series)
+        {
+            movie = _movieService.FindByImdbId(imdbId);
+            //Should fix practically all problems, where indexer is shite at adding correct imdbids to movies.
+            if (movie != null && parsedMovieInfo.Year > 1800 && parsedMovieInfo.Year != movie.Year)
+            {
+                movie = null;
+                return false;
+            }
+            return movie != null;
+        }
+
+        private bool TryGetMovieByTitleAndOrYear(ParsedMovieInfo parsedMovieInfo, out Movie movieByTitleAndOrYear)
+        {
+            Func<Movie, bool> isNotNull = movie => movie != null;
+            if (parsedMovieInfo.Year > 1800)
+            {
+                movieByTitleAndOrYear = _movieService.FindByTitle(parsedMovieInfo.MovieTitle, parsedMovieInfo.Year);
+                if (isNotNull(movieByTitleAndOrYear))
+                {
+                    return true;
                 }
+            }
+            movieByTitleAndOrYear = _movieService.FindByTitle(parsedMovieInfo.MovieTitle);
+            if (isNotNull(movieByTitleAndOrYear))
+            {
+                return true;
+            }
+            movieByTitleAndOrYear = null;
+            return false;
+        }
 
-				if (possibleMovie != null && (parsedMovieInfo.Year < 1800 || possibleMovie.Year == parsedMovieInfo.Year))
-				{
-					return possibleMovie;
-				}
+        private static bool TryGetMovieBySearchCriteria(ParsedMovieInfo parsedMovieInfo, SearchCriteriaBase searchCriteria, out Movie possibleMovie)
+        {
+            possibleMovie = null;
+            var possibleTitles = new List<string>();
+            Movie possibleMovieBySearchCriteria = null;
 
-                    
-            }
+            possibleTitles.Add(searchCriteria.Movie.CleanTitle);
+            var arabicRomanNumeralMappings = RomanNumeralParser.GetArabicRomanNumeralsMapping();
 
-            Movie movie = null;
+            foreach (string altTitle in searchCriteria.Movie.AlternativeTitles)
+            {
+                possibleTitles.Add(altTitle.CleanSeriesTitle());
+            }
 
-            if (searchCriteria == null)
+            foreach (string title in possibleTitles)
             {
-                if (parsedMovieInfo.Year > 1800)
-                {
-                    movie = _movieService.FindByTitle(parsedMovieInfo.MovieTitle, parsedMovieInfo.Year);
-                }
-                else
+                if (title == parsedMovieInfo.MovieTitle.CleanSeriesTitle())
                 {
-                    movie = _movieService.FindByTitle(parsedMovieInfo.MovieTitle);
+                    possibleMovieBySearchCriteria = searchCriteria.Movie;
                 }
 
-                if (movie == null)
+                foreach (ArabicRomanNumeral numeralMapping in arabicRomanNumeralMappings)
                 {
-                    movie = _movieService.FindByTitle(parsedMovieInfo.MovieTitle);
```

Comment:
After taking a second glance I now understand why the same method with the same parameters is seemingly called twice for no goo reason. It might happen that line 406 evaluates true but no movie is found. The else block would never be called in case the movie is null. In my PR I have simply deleted the seemingly redundant call to `FindByTitle(..)` which was a mistake. I will adjust it accordingly but in way that is more expressive/explicit.